packages/forma-36-react-components/src/components/NavigationIcon/NavigationIcon.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { Component, CSSProperties } from 'react';
+import cn from 'classnames';
+import { navigationIconName } from './constants';
+
+import Spaces from './svg/Spaces.svg';
+import Apis from './svg/Apis.svg';
+import Billing from './svg/Billing.svg';
+import Home from './svg/Home.svg';
+import Media from './svg/Media.svg';
+import ContentModel from './svg/ContentModel.svg';
+import Content from './svg/Content.svg';
+import Apps from './svg/Apps.svg';
+import Oauth from './svg/Oauth.svg';
+import OrgInfo from './svg/OrgInfo.svg';
+import Oraganizations from './svg/Oraganizations.svg';
+import Settings from './svg/Settings.svg';
+import Sso from './svg/Sso.svg';
+import Subscription from './svg/Subscription.svg';
+import Teams from './svg/Teams.svg';
+import Token from './svg/Token.svg';
+import Usage from './svg/Usage.svg';
+import UserProfile from './svg/UserProfile.svg';
+import Users from './svg/Users.svg';
+
+import styles from './NavigationIcon.css';
+
+const iconComponents = {
+  Spaces,
+  Apis,
+  Apps,
+  Billing,
+  ContentModel,
+  Content,
+  Home,
+  Media,
+  Oauth,
+  OrgInfo,
+  Oraganizations,
+  Settings,
+  Sso,
+  Subscription,
+  Teams,
+  Token,
+  Usage,
+  UserProfile,
+  Users,
+};
+
+export type NavigationIconType = keyof typeof navigationIconName;
+export type NavigationIconColorType =
+  | 'primary'
+  | 'positive'
+  | 'negative'
+  | 'warning'
+  | 'secondary'
+  | 'muted'
+  | 'white';
+
+export type NavigationIconSize = 'small' | 'medium' | 'large' | 'xlarge';
+export type NavigationIconTagType = 'div' | 'span';
+
+export interface NavigationIconProps {
+  size?: NavigationIconSize;
+  color?: NavigationIconColorType;
+  style?: CSSProperties;
+  icon: NavigationIconType;
+  className?: string;
+  testId?: string;
+  tagType?: NavigationIconTagType
+}
+
+const defaultProps = {
+  testId: 'cf-ui-navigation-icon',
+  size: 'medium',
+  color: 'positive',
+  tagType: 'div'
+};
+
+export class NavigationIcon extends Component<NavigationIconProps & typeof defaultProps> {
+  static defaultProps = defaultProps;
+  
+  render() {
+    const { className, icon, size, color, testId, tagType, ...otherProps } = this.props;
+    
+    const classNames = cn(
+      styles.NavigationIconContainer,
+      {
+        [styles[`NavigationIconContainer--${size}`]]: size,
+        [styles['NavigationIconContainer--trimmed']]: icon.toLowerCase().includes('trimmed'),
+      },
+      className,
+    );
+
+    const svgFillClass = cn(
+      styles.NavigationIcon,
+      {
+        [styles[`NavigationIcon--${color}`]]: color,
+      },
+      className,
+    )
+    const Element = iconComponents[icon];
+    const Tag = tagType;
+
+    return (
+      <Tag
+        className={classNames}
+        data-test-id={testId}
+      >
+        <Element  className={svgFillClass} {...otherProps} />
+      </Tag>
```

Comment:
I think I didn't understand the <Tag> here, it can be a div or a span. But do we need to create it with this wrapper? I looked into the Icon component and it does not have it

Reply:
@gui-santos yeah, good point. I thought that it might be sligthly different in this case - the container controll the size of the icon, SVG itself knows only about the the color. when implementing we would need to create wrapper on the application side anyway I guess, so I figured that maybe it could live in the component itself? This is how I implemented this component in the app, but it can totally be changed if you fill like that's not the right approach. We can go with what we have in the Icon 1:1.

Reply:
Ohhh I get it now!
Thanks for the explanation. I think it makes sense, so IMO you can leave it like this :)

cool stuff